### ISSUE TITLE
[0.25] Tracking Cookie for Lifetime Member Referral Program

### ISSUE THREAD

**Issue by bangzi1001** [bitshares/bitshares-ui#2555]
**Is your feature request related to a problem? Please describe.**
When lifetime member Mr. Apple bring new user Mr. Banana to Bitshares reference wallet via referral link eg. https://wallet.bitshares.org/?r=apple, the referral program only work if the Mr. Banana register within the session. If Mr. Banana visit and leave without register but come back next day via direct type in wallet.bitshares.org or Google search, Mr. Apple no longer become Mr. Banana referrer. This strongly demotivate lifetime members to promote Bitshares.

**Describe the solution you'd like**
Add tracking cookie that expire after 180 days once Mr. Banana visit reference wallet via referral link. Within 180 Days, if Mr. Banana visit Bitshares reference wallet again and register an account, Mr. Apple will become his referrer. 

*This feature is important for Bitshare Marketing and it complement BSIP43: Market Fee Sharing

**Comment by startailcoon** [comment]
To allow a cookie we need to inform the visitor that we do so, and also enable the visitor to deny using cookies as per EU legislation on cookies, http://ec.europa.eu/ipg/basics/legal/cookies/index_en.htm#section_2

**Comment by startailcoon** [comment]
With the new market fee sharing, we should make this work better. Some users will be doing campaigns to drive more new users to the platform, and without a proper session storage for the account that referred them, the risk that these campaigns will give up are high.

Today we store a referral in the Local storage under `__graphene_referralAccount`. For some reason, this isn't saved over sessions.

**Comment by startailcoon** [comment]
Pushed an update to store referral over sessions.

Previously we removed store referral if none was appended to the visited URL. If a user where returning the next day, it would not persist, unless they used the ref in the address again.

I chose to make it check if we have a previously stored referral, and only store it if none has previously been submitted. This would make sure the original referrer is used. If a user comes to the page with a new referrer, it would still use the old one.
